### PR TITLE
Change various depends_on to uses_from_macos

### PR DESCRIPTION
**Pull request by FnControlOption**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Working on a new audit that checks if `depends_on` inside `on_linux` can be `uses_from_macos`

### CODE REVIEW

File: Formula/swift.rb [hunk 1]
```diff
@@ -40,6 +40,7 @@ class Swift < Formula
   uses_from_macos "llvm" => [:build, :test]
   uses_from_macos "rsync" => :build
   uses_from_macos "curl"
+  uses_from_macos "icu4c" # Used in swift-corelibs-foundation
```
Comment (carlocab) [inline Formula/swift.rb:43]: I think the macOS build doesn't use `libicu` here.

Comment (Bo98) [reply Formula/swift.rb:43]: Yes, that's correct. swift-corelibs-foundation is only built on Linux, so the Linux-only `depends_on` is semantically more correct here.

Comment (FnControlOption) [reply Formula/swift.rb]: I added `swift` to `provided_by_macos_depends_on_allowlist.json` but do you think a new allowlist should be created?

**Review by MikeMcQuaid — APPROVED** [review]
Very nice 👏🏻

**Pull request closed without merging**